packages/grid/x-data-grid-pro/src/tests/editRows.DataGridPro.test.tsx

```diff
@@ -1520,6 +1521,34 @@ describe('<DataGridPro /> - Edit Rows', () => {
         expect(secondInput).to.have.attribute('aria-invalid', 'true');
       });
     });
+
+    it('should exit the row edit mode and save the row when preProcessEditCellProps does not return an error', async () => {
+      render(
+        <TestCase
+          editMode="row"
+          columns={[
+            {
+              field: 'brand',
+              editable: true,
+              preProcessEditCellProps: ({ props }) => props,
+            },
+          ]}
+          rows={[{ id: 0, brand: 'Nike' }]}
+        />,
+      );
+      const cell = getCell(0, 0);
+      fireEvent.mouseUp(cell);
+      fireEvent.doubleClick(cell);
+      const input = cell.querySelector('input')!;
+      fireEvent.change(input, { target: { value: 'Adidas' } });
+      await clock.runToLast();
```

Comment:
Why the `await` here? This test is not passing on Safari.

Reply:
I spent quite some time to make this test work.
It did not on Chrome Headless without the `await`
But if the `await` causes problem then the solution is not good.